### PR TITLE
Writing style updates to Developing dynamic inventory topic

### PR DESCRIPTION
**Pull request by sayee-jadhav**
SUMMARY
Incorporates writing style changes per the IBM style guide.

ISSUE TYPE
Docs Pull Request

COMPONENT NAME
docs.ansible.com
inventory

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 1]
```diff
@@ -4,30 +4,27 @@
...
+
+.. contents:: Topics
+   :local:
 
 
 .. _inventory_sources:
 
-Inventory sources
-=================
+What are inventory sources?
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]: I'd undo this change.

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 2]
```diff
@@ -48,17 +45,17 @@ The table below shows some examples of inventory plugins and the kinds of source
 
 .. _developing_inventory_inventory_plugins:
 
-Inventory plugins
-=================
+What are inventory plugins?
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]: Here too, I'd avoid using a question for the header.

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 3]
```diff
@@ -48,17 +45,17 @@ The table below shows some examples of inventory plugins and the kinds of source
...
 
 Most of the documentation in :ref:`developing_plugins` also applies here. You should read that document first for a general understanding and then come back to this document for specifics on inventory plugins.
 
-Inventory plugins normally only execute at the start of a run, before playbooks, plays, and roles are loaded.
-However, you can use the ``meta: refresh_inventory`` task to clear the current inventory and to execute the inventory plugins again, which will generate a new inventory.
+Normally, inventory plugins are executed at the start of a run, and before the playbooks, plays, or roles are loaded.
+However, you can use the ``meta: refresh_inventory`` task to clear the current inventory and execute the inventory plugins again, and this task will generate a new inventory.
 
-If you use the persistent cache, inventory plugins can also use the configured cache plugin to store and retrieve data. This avoids repeating costly external calls.
+If you use the persistent cache, inventory plugins can also use the configured cache plugin to store and retrieve data. This data retrieval avoids making repeated and costly external calls.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]:
```suggestion
If you use the persistent cache, inventory plugins can also use the configured cache plugin to store and retrieve data. Caching inventory avoids making repeated and costly external calls.
```

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 4]
```diff
@@ -153,7 +147,7 @@ The base class does some minimal assignment for reuse in other methods.
         self.inventory = inventory
         self.templar = Templar(loader=loader)
 
-It is up to the plugin now to deal with the inventory source provided and translate that into the Ansible inventory.
+It is upto the plugin now to parse the provided inventory source and translate it to Ansible inventory.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]: Typo!
```suggestion
It is up to the plugin now to parse the provided inventory source and translate it into Ansible inventory.
```

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 5]
```diff
@@ -272,25 +266,25 @@ Now that you've enabled caching, loaded the correct plugin, and retrieved a uniq
...
+Depending on other common features used, you might need other fields, and you can add custom options in each plugin as required.
+For example, if you use the integrated caching, ``cache_plugin``, ``cache_timeout`` and other cache-related fields could be present.
 
 .. _inventory_development_auto:
 
 The 'auto' plugin
 -----------------
 
-Since Ansible 2.5, we include the :ref:`auto inventory plugin <auto_inventory>` enabled by default, which itself just loads other plugins if they use the common YAML configuration format that specifies a ``plugin`` field that matches an inventory plugin name, this makes it easier to use your plugin w/o having to update configurations.
+From Ansible 2.5 onwards, we include the :ref:`auto inventory plugin <auto_inventory>` that is enabled by default. This plugin loads other plugins that use the common YAML configuration format and specifies a ``plugin`` field with a matching inventory plugin name. The 'auto' plugin makes it easier to use your plugin without having to update configurations.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]:
```suggestion
From Ansible 2.5 onwards, we include the :ref:`auto inventory plugin <auto_inventory>` and enable it by default. If the ``plugin`` field in your standard configuration file matches the name of your inventory plugin, the ``auto`` inventory plugin will load your plugin. The 'auto' plugin makes it easier to use your plugin without having to update configurations.
```

File: docs/docsite/rst/dev_guide/developing_inventory.rst [hunk 6]
```diff
@@ -334,17 +326,17 @@ and potential group variables, or simply a list of hosts::
...
+When called with the argument ``--host <hostname>`` (where <hostname> is a host from above), the script must print either an empty JSON hash/dictionary, or a hash/dictionary of variables to make them available to templates and playbooks. For example::
 
 
     {
         "VAR001": "VALUE",
         "VAR002": "VALUE",
     }
 
-Printing variables is optional. If the script does not do this, it should print an empty hash or dictionary.
+Printing variables is optional. If the script does not do it, the script should print an empty hash or dictionary.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_inventory.rst]:
```suggestion
Printing variables is optional. If the script does not print variables, it should print an empty hash or dictionary.
```

**Review by acozine — COMMENTED** [review]
This looks great, thanks @sayee-jadhav. I found a few places where we could make it even better.

**Comment by sayee-jadhav** [comment]
Thank you for the review, @acozine  +1 to your suggestions. I updated the source file and committed the changes. Request you to please merge this PR.
